code/src/store/api/productsApi.ts

```diff
@@ -0,0 +1,35 @@
+import { appApi } from "@/store/api/appApi";
+import { httpMethods } from "@/constants/methods";
+import { createUrlPath } from "@/utils/createUrlPath";
+import { URLS } from "@/constants/requests";
+import { Product } from "@/types/product.types";
+
+const productsApi = appApi.injectEndpoints({
+  endpoints: (build) => ({
+    getProducts: build.query<Product[], void>({
+      query: () => URLS.products.get
+    }),
+    addProduct: build.mutation<
+      any,
+      { name: string; description: string; price: number }
```

Comment:
You already have `Product` type. Better use Pick<Product, "name" | "description" | "price">.

Reply:
Fixed